### PR TITLE
Entry delay and effect

### PR DESCRIPTION
**Pull request by kirderf1**
Puts the entry at a delay from when the land dimension is generated to when the actual entry happens. This gives land worldgen a couple of ticks to do its thing, ideally improving the performance of entry. Since how much time worldgen needs depends on the performance of the machine running it, this delay is configurable in the common config.
To ensure that the game tries to load the chunks (and keeps them loaded), a chunk ticket gets added that is removed once entry is done.

An effect has been added that spawns light-blue particles in the entry area during this time.

Playtesting of this would be appreciated, at the very least to see if the current default value is suitable or should be higher, as well as if the entry effect should be adjusted in some way.

I've also considered a command that triggers land generation + chunk ticket earlier than that (for GMs to use), but I might wait on that for a later time unless someone wants me to add it with this PR.

### CODE REVIEW

**Comment by Dweblenod** [comment]
I will admit it sound counter intuitive to what this PR accomplishes, but with the improvements to performance and with the lower world generation limit do you think it would make sense to increase the depth/total blocks that players Enter with?

**Comment by kirderf1** [comment]
I did consider it, though at the moment I think we should hold off and see how things works out with this delay first.

**Comment by Dweblenod** [comment]
> I did consider it, though at the moment I think we should hold off and see how things works out with this delay first.

Makes sense!

File: src/main/java/com/mraof/minestuck/MinestuckConfig.java
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/MinestuckConfig.java:39]: Recommendation to double the default value

**Review by Dweblenod — COMMENTED** [review]
First round of testing went well! Tried double entry, getting a bunch of moving blocks to see if there would be any duplications, etc. Next will try it with a dedicated server

File: src/main/java/com/mraof/minestuck/MinestuckConfig.java
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/MinestuckConfig.java:39]: Actually perhaps even more. That way performance is optimal and players get more of a chance to see the particle effects

**Comment by Dweblenod** [comment]
I may be misunderstanding it's use, but perhaps the MIN_TIME field in PostEntryTask should also be increased from 15 to 20 to improve the post entry strain on the server

**Comment by kirderf1** [comment]
Ah, I'll add a piece of javadoc for that

**Comment by kirderf1** [comment]
Or no, see over the javadoc that's already there

**Comment by Dweblenod** [comment]
> Or no, see over the javadoc that's already there
Ah! Then perhaps it should be lowered from 15 to 10 to reduce strain

File: src/main/java/com/mraof/minestuck/entry/EntryProcess.java
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/entry/EntryProcess.java:158]: In testing, its likely the player will reenter underground

**Review by Dweblenod — APPROVED** [review]
Besides from the second entry placement, didnt find any bugs so far in a dedicated server or in single player!

I created a PR for a recommendation I have been thinking of if you would be willing to look at it before you merge this but other than that and the consideration to implement custom particle effects as discussed on Discord it seems good!

**Review by Dweblenod — CHANGES_REQUESTED** [review]
Now that players can move around in the period before entry actually occurs, they have the ability to move elsewhere. If they move out of the entry region they may fall to their death or suffocate when they are teleported with the land as they will teleport to a spot off the entry platform

File: src/main/java/com/mraof/minestuck/entry/EntryProcess.java
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/entry/EntryProcess.java]: Has now been changed to use the heightmap for motion blocking blocks.

**Comment by Dweblenod** [comment]
Doing another in-game test now

**Review by Dweblenod — APPROVED** [review]

**Comment by Dweblenod** [comment]
Seems good to go!

**Pull request merged**
